packages/mui-material/src/InputLabel/InputLabel.spec.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import InputLabel from '@mui/material/InputLabel';
+import { expectType } from '@mui/types';
+
+const CustomComponent: React.FC<{ prop1: string; prop2: number }> = function CustomComponent() {
+  return <div />;
+};
+
+const InputLabelTest = () => {
+  return (
+    <div>
+      <InputLabel />
+      <InputLabel component="legend" />
+      <InputLabel
+        component="legend"
+        onClick={(event) => {
+          expectType<React.MouseEvent<HTMLLabelElement, MouseEvent>, typeof event>(event);
```

Comment:
Shouldn't this be `HTMLLegendElement` type now that the underlying element is `legend`?

Reply:
Actually you are right StandardProps is not required as all props of FormLabel is already present in InputLabel and StandardProps is was causing event type to be infered as `HTMLLabelElement`. i fixed type inferring by removing StandardProps.

Added test :

https://github.com/mui/material-ui/pull/35852/files#diff-f5beb3d0a6cd307137bc2c29e830648a962b649fb44ea6b0221f9d0360f8cadaR14-R17

Reply:
okay so `StanardProps<FormLabelProps>` are overriding `component` event handlers types, at first i removed `StanardProps<FormLabelProps> `and everything worked great but  looking at `InputLabel` api docs (https://mui.com/material-ui/api/input-label/#props) it says `Props of the FormLabel component are also available.` so may there could be reason behind extending StanardProps<FormLabelProps>.

So i solved event handler overriding behaviour by omiting `FormLabelBaseProps` keys and now everthing working great.

added test to make sure correct type is set to event handlers.

https://github.com/mui/material-ui/pull/35852/files#diff-f5beb3d0a6cd307137bc2c29e830648a962b649fb44ea6b0221f9d0360f8cadaR15-R18

Reply:
What I meant by removing `StandardProps` was to remove the `StandardProps` type but keep the extension of the `FormLabelProps` i.e
```diff
- extends StandardProps<FormLabelProps> {
+ extends FormLabelProps {
```
Sorry, I should have been clear. Anyway, see https://github.com/mui/material-ui/pull/35852#pullrequestreview-1265204672.